snapcraft/plugins/_plugin_finder.py

```diff
@@ -86,10 +86,15 @@ def get_plugin_for_base(plugin_name: str, *, build_base: str) -> PluginTypes:
     """
     # TODO: segregate this more and remove the checks UnsupportedBase checks
     # from the plugins
-    if build_base in ("core", "core16", "core18"):
-        build_base = "legacy"
+
+    # Default for unknown, and core20
+    plugin_version = "v2"
+    # Others use v1
+    if build_base in ("", "legacy", "core", "core16", "core18"):
```

Comment:
Seems that legacy faced a copy / paste error
```suggestion
    if build_base in ( "core", "core16", "core18"):
        plugin_version = "v1"
    elif build_base == "":
        # This should never happen if 
        raise RuntimeError("'build_base' cannot be unset.")
```

The last check is a precaution, in case someone skips this other precaution in `snapcraft.internal.meta.snap.Snap.get_build_base` which is only triggered if someone skips schema checks during development:
```python
# The schema does not allow for this when loaded from snapcraft.yaml.
        if build_base is None:
            raise RuntimeError("'build_base' cannot be None")
```